### DISCUSSION TITLE
Allow to ignore the last X minor or major releases

### DISCUSSION THREAD

**Discussion by quentinus95** [renovatebot/renovate#24837] [Suggest an Idea]
### What would you like Renovate to be able to do?

The aim of this feature is to help to ignore newer versions of a dependency or package that might provide a decrease of stability.

This is particularly useful when having dependencies that:

- release a lot of new versions in a short amount of time (e.g., an NPM dependency that release a patch every week, and a minor every month)
- have monthly releases (e.g., GitLab releases a new version every month. It can be useful to upgrade to the last patch of the previous release every month instead of the new minor right away)
- are critical dependencies (e.g., always ignoring the last two major releases of Kubernetes while benefiting from the last patches of the previous majors).

Today, the number of updates can be overwhelming. Having a way to ignore most of the consecutive patches and always update to the last patch of the previous minor can provide a good trade-off between last features and quick fixes.

Packages sometimes provide maintenance rules for multiple versions at the same time (e.g., Symfony might release a patch for critical security issues over multiple majors). With this new configuration solution you can still benefit from those fixes. You may also sometimes need to plan for extensive work when upgrading to a major version. For such situations, Renovate might not be very useful. You know that you need to update to a major, it will require several weeks for your team to upgrade. You don't want to be flooded by Renovate for each patch on this version.

Finally, you can combine both options. For instance, you may want to ignore the last major and the last minor of a specific package (e.g., Kubernetes).

### If you have any ideas on how this should be implemented, please tell us here.

Add two new configuration options:
- `ignoreLastXMajor`: ignore the last X major versions 
- `ignoreLastXMinor`: ignore the last X minor versions

See https://github.com/renovatebot/renovate/pull/18218

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by rarkins** [comment]
The challenge with something like this is making sure we think of the edge conditions. On one hand, this feature would be opt-in, so nobody's forcing the user to use it. On the other hand, users can be like this:

![image](https://user-images.githubusercontent.com/6311784/195014430-59f8a2df-3a7b-4687-99bf-0a492784995d.png)

**Comment by quentinus95** [comment]
Thanks, I would be happy to update the implementation to cover such cases. If you are aware of any other, please share as well.
I do not remember if the vulnerability fix information was present in the release info when I was filtering them. I will double-check. If not do you know how I can retrieve it?

**Comment by rarkins** [comment]
This is how we try to "bypass" controls for vulnerability alerts currently: https://github.com/renovatebot/renovate/blob/c3a87b687ed44a1a100e0e4f91b81e22b7c32482/lib/config/options/index.ts#L1598-L1607

**Comment by rarkins** [comment]
For ignoreLastXMajor, how would this be better than the following?
- separateMultipleMajor=true
- dependencyDashboardApproval=true

With this, you'll never get a major update PR automatically, but you will see them in the dashboard. You'll still get non-major PRs by default.

**Comment by quentinus95** [comment]
@rarkins I agree, it would be ok for the ignoreLastXMajor. The main use case I want to solve is more for minor releases, that happen often and that are more likely to break applications. Having a way to always wait for the next minor to benefit from the patches is a way to improve applications robustness.
